client/asset/btc/spv_peer_manager.go

```diff
@@ -0,0 +1,236 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package btc
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"path/filepath"
+	"sync"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/client/asset/kvdb"
+	"decred.org/dcrdex/dex"
+)
+
+type peerSource uint16
+
+const (
+	added peerSource = iota
+	defaultPeer
+)
+
+func (s peerSource) toAssetPeerSource() asset.PeerSource {
+	switch s {
+	case added:
+		return asset.UserAdded
+	case defaultPeer:
+		return asset.WalletDefault
+	}
+	return asset.Discovered
+}
+
+func (s peerSource) MarshalBinary() ([]byte, error) {
+	b := make([]byte, 16)
+	binary.BigEndian.PutUint16(b, uint16(s))
+	return b, nil
+}
+
+type walletPeer struct {
+	source peerSource
+	// resolvedNames is used to avoid adding duplicate peers
+	resolvedNames []string
+}
+
+// PeerManagerChainService are the functions needed for an SPVPeerManager
+// to communicate with a chain service.
+type PeerManagerChainService interface {
+	AddPeer(addr string) error
+	RemoveNodeByAddr(addr string) error
+	Peers() []SPVPeer
+}
+
+// SPVPeerManager implements peer management functionality for all bitcoin
+// clone SPV wallets.
+type SPVPeerManager struct {
+	cs PeerManagerChainService
+
+	peersMtx sync.RWMutex
+	peers    map[string]*walletPeer
+	peersDB  kvdb.KeyValueDB
+
+	defaultPeers []string
+
+	log dex.Logger
+}
+
+// Peers returns the list of peers that the wallet is connected to. It also
+// returns the peers that the user added that the wallet may not currently
+// be connected to.
+func (w *SPVPeerManager) Peers() ([]*asset.WalletPeer, error) {
```

Comment:
Why `w`?

Reply:
I'll update, I had these methods on the btcSPVWallet type earlier.